scripts/CSSRE/get-node-restarts/script.py

```diff
@@ -0,0 +1,137 @@
+'''
+CheckNodeRestart is a managed backplane script in order to
+find out the NODES that got restarted in the LAST_MIN duration
+and list out all the pods that the associated with the particular NODE
+
+'''
+import argparse
+import json
+import datetime
+import sys
+
+LIB_PATH = "/managed-scripts/CSSRE/lib"
+# Use sys.path.insert because PYTHONPATH's support is unclear
+sys.path.insert(0, LIB_PATH)
+
+from py.script import Script
+
+LAST_MIN = 60 # default value
+
+class CheckNodeRestart(Script):
+    """
+    CheckNodeRestart is a managed backplane script to find out
+    NODES that got restarted in the LAST_MIN duration
+
+    Usage:
+    ocm backplane managedjob create # with default env variables
+    ocm backplane managedjob create -p LAST_MIN=<duration> -p LOG_LEVEL=""
+    """
+
+    def __init__(self):
+        # this will ensure optional environment variables
+        env_vars = ["LAST_MIN", "LOG_LEVEL"]
+        self.parser = None
+        super().__init__(env_vars=env_vars, check_env_var=False)
+
+    def create_parser(self):
+        self.parser = argparse.ArgumentParser()
+        # self.parser.add_argument(
+        #     "last_min", type=int, help="NODES restarted in the last X min"
+        # )
+        # self.parser.add_argument(
+        #     "log_level", nargs="?", default="info"
+        # )
+
+    def get_node_data(self):
+        """
+        Get node event metadata
+        """
+        _, node_data, errors = self.cmd.run_pipe(["oc", "get", "nodes", "-o", "json"], check=False)
+        if errors:
+            self.logger.error(errors)
+            self.exit(1)
+        node_data = json.loads(node_data)
+        return node_data
+
+    @staticmethod
+    def get_node_transition_ts(node_data):
+        """
+        Get Last Transition time of each node for
+        reason 'KubeletReady'
+        """
+        output_dict = {}
+        nodes = node_data.get("items", [])
+        if nodes:
+            for node in nodes:
+                conditions = node["status"]["conditions"]
+                for condition in conditions:
+                    if condition["reason"] == "KubeletReady":
+                        output_dict[node["metadata"]["name"]] = condition[
+                            "lastTransitionTime"
+                        ]
```

Comment:
output_dict[node["metadata"]["name"]] = condition[ "lastTransitionTime"]

Reply:
do you mean that this should be a oneliner? if so it might be the IDE that automatically does this
you could also
```
transition_condition = condition["lastTransitionTime"]
output_dict[node["metadata"]["name"]] = transition_condition

Reply:
yup, oneliner

Reply:
@wanghaoran1988 this is done https://github.com/openshift/managed-scripts/pull/60/commits/2cc35298b8f3d5cb41d6e13e1e26d43ef98cf32e

Reply:
@wanghaoran1988 @georgettica can you please look at this change and merge this PR